internal/integration/images_test.go

```diff
@@ -0,0 +1,222 @@
+			"url":               "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest/rhcos-live.x86_64.iso",
+			"rootfs_url":        "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest/rhcos-live-rootfs.x86_64.img",
+		},
+	}
+
+	assistedServer       *ghttp.Server
+	imageIDWithInitRD    = "bf25292a-dddd-49dc-ab9c-3fb4c1f07071"
+	imageIDWithoutInitRD = "bf25292a-dddd-49dc-ab9c-3fb4c1f07072"
+	ignitionContent      = []byte("someignitioncontent")
+	initrdContent        = []byte("someramdisk")
+	imageServer          *httptest.Server
+	imageClient          *http.Client
+
+	imageDir        string
+	scratchSpaceDir string
+	is              imagestore.ImageStore
+	ctxBg           = context.Background()
+)
+
+func verifyIgnition(fs filesystem.FileSystem) {
+	// TODO(djzager): Export isoeditor.ignitionImagePath?
+	f, err := fs.OpenFile("/images/ignition.img", os.O_RDONLY)
+	Expect(err).NotTo(HaveOccurred())
+
+	gzipReader, err := gzip.NewReader(f)
+	Expect(err).NotTo(HaveOccurred())
+	cpioReader := cpio.NewReader(gzipReader)
+	hdr, err := cpioReader.Next()
+	Expect(err).NotTo(HaveOccurred())
+	// TODO(djzager): Tie this to some constant?
+	Expect(hdr.Name).To(Equal("config.ign"))
+	Expect(hdr.Size).To(Equal(int64(len(ignitionContent))))
+
+	content, err := ioutil.ReadAll(cpioReader)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(content).To(Equal(ignitionContent))
+}
+
+func verifyRamdisk(fs filesystem.FileSystem, expectedContent []byte) {
+	// TODO(djzager): Should we export isoeditor.ramDiskImagePath?
+	f, err := fs.OpenFile("/images/assisted_installer_custom.img", os.O_RDONLY)
+	Expect(err).NotTo(HaveOccurred())
+
+	content, err := ioutil.ReadAll(f)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(bytes.TrimRight(content, "\x00")).To(Equal(expectedContent))
+}
+
+func getImage(path, imageType string, version map[string]string) filesystem.FileSystem {
+	resp, err := imageClient.Get(imageServer.URL + path)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+	isoPath := filepath.Join(scratchSpaceDir, fmt.Sprintf("test-%s-%s.%s.iso", version["openshift_version"], imageType, version["cpu_architecture"]))
+	out, err := os.Create(isoPath)
+	Expect(err).NotTo(HaveOccurred())
+	_, err = io.Copy(out, resp.Body)
+	Expect(err).NotTo(HaveOccurred())
+
+	d, err := diskfs.OpenWithMode(isoPath, diskfs.ReadOnly)
+	Expect(err).NotTo(HaveOccurred())
+	fs, err := d.GetFilesystem(0)
+	Expect(err).NotTo(HaveOccurred())
+
+	return fs
+}
+
+var _ = Describe("Image integration tests", func() {
+	Context("full-iso", func() {
+		BeforeEach(func() {
+			assistedServer.AppendHandlers(
+				ghttp.CombineHandlers(
+					ghttp.VerifyRequest("GET", fmt.Sprintf("/api/assisted-install/v2/infra-envs/%s/downloads/files", imageIDWithInitRD), "file_name=discovery.ign"),
+					ghttp.RespondWith(http.StatusOK, ignitionContent),
+				),
+			)
+		})
+
+		for _, version := range versions {
```

Comment:
I _think_ this runs us into the for-loop address issue.
Can you make sure we're not just testing the same version multiple times?

Reply:
This applies to all the places we have a range loop followed directly by an `It`

Reply:
This is still unresolved...or at least unverified.

Reply:
Verified this is fixed with latest change.